### ISSUE TITLE
modpack with optifine error

### ISSUE THREAD

**Issue by StenaLeonny** [NillerMedDild/Enigmatica4#386]
the modpack does not allow me to use optifine every time i put optifine in the modpack it before reaching the loading screen it goes down and this message appears in the error;

The game crashed whilst rendering overlay
Error: java.lang.NullPointerException: Rendering overlay

**Comment by NielsPilgaard** [comment]
Optifine does not work with Forge for Minecraft 1.14.4

**Comment by StenaLeonny** [comment]
I play other 1.14.4 modpack. And optifine works.    If you want I can send print to show you.

**Comment by NielsPilgaard** [comment]
Which modpacks? :)

**Comment by Darkere** [comment]
Optifine did work with Forge until somewhere around Version 28.2.0. Still haven't from anyone if downgrading forge works....

**Comment by Darkere** [comment]
Ok. Downgrading forge does not work with 0.5.4 as multiple mods require the current forge version.

**Comment by StenaLeonny** [comment]
I play a modpack in version 1.14.4 called janky skyblock and this modpack uses forge 28.1.96 and optifine works perfectly

**Comment by NielsPilgaard** [comment]
Alright, let me rephrase: Optifine does not work with recent versions of Forge for 1.14.4 :P

**Comment by StenaLeonny** [comment]
But do you know if you can do it?  Maybe in a new version that comes from the forge?

**Comment by NielsPilgaard** [comment]
It's not something I can currently do, no. Why do you want Optifine btw?

**Comment by StenaLeonny** [comment]
I can play the modpack without optifine but it crashes a lot and I don't get out of 50 pfs. with optifine i can remove the particles and change the settings in order to go from 50 fps to 100+ fps

**Comment by NielsPilgaard** [comment]
Please report crashes you run into so we can fix them :)

**Comment by tyrstag** [comment]
I am also running into the same issue with OptiFine, but my reasons for wanting it are related to shadders in the environment. Just personal preference. 
Attached is my recent crash report

[crash-2020-04-30_11.45.13-client.txt](https://github.com/NillerMedDild/Enigmatica4/files/4559463/crash-2020-04-30_11.45.13-client.txt)

//Edit - OptiFine is at fault for compatibility issues with Forge. No version of Optifine will work with any version of Forge after 28.1.56

**Comment by theloser** [comment]
> No version of Optifine will work with any version of Forge after 28.1.56

Wrong. I've got MC 1.14.4 running with Forge 28.2.0 (which is the recommended version for 1.14.4!) and OptiFine 1.14.4_HD_U_F5 -- and it works perfectly (even with ~100 mods and Shaders)!

I'm curious, which mods from Enigmatica 4 require a higher version of Forge than the recommended version?

EDIT: Nevermind, I changed my Forge to version 28.2.0 and I can see which mods require a higher Forge version. It's only three mods, though... (as of Enigmatica 4 v0.5.4).

**Comment by tyrstag** [comment]
> //Edit - OptiFine is at fault for compatibility issues with Forge. No version of Optifine will work with any version of Forge after 28.1.56
  
Statement based off of changelogs on optifine's site. A more appropriate statement would have been, "no version of Optifine is stated as supporting any version of Forge after 28.1.56" I wonder why yours is working. Just normal install for your other pack?

**Comment by NielsPilgaard** [comment]
Optifine should now work on more recent Forge for 1.14.4, given you add this mod yourself: https://www.curseforge.com/minecraft/mc-mods/optiforge

<blockquote><div><strong><a href="https://www.curseforge.com/minecraft/mc-mods/optiforge">Attention Required! | Cloudflare</a></strong></div></blockquote>

**Comment by theloser** [comment]
@tyrstag :
> I wonder why yours is working. Just normal install for your other pack?

You're right, there's no official support for Forge 28.2.x by OptiFine. I installed it anyway (Forge 28.2.0) and it works perfectly. This is no specific Modpack, it's just a handpicked selection of mods from Enigmatica 4 since I couldn't get this Modpack to work with Optifine.

@NillerMedDild :
> Optifine should now work on more recent Forge for 1.14.4, given you add this mod yourself

Thanks a lot for this info! I just tried it and it works (at least it's not crashing right at the start). If only I'd knew that Shaders and Nightvision don't go together very well... ;-)

EDIT: Can't recommend the OptiForge solution! After playing a while (even without shaders) all fonts get messed up, and you can't read anything (tooltips, menus etc.).

**Comment by NielsPilgaard** [comment]
> EDIT: Can't recommend the OptiForge solution! After playing a while (even without shaders) all fonts get messed up, and you can't read anything (tooltips, menus etc.).

That is one of the many reasons we don't support Optifine, it causes unexpected behaviour and bugs.

**Comment by theloser** [comment]
> That is one of the many reasons we don't support Optifine, it causes unexpected behaviour and bugs.

I didn't say it was OptiFine's fault, I blamed Opti**Forge** for the bug! As said earlier, OptiFine works flawless on my secondary MC-1.14.4-Forge-28.2.0 install.
